quinn-proto/src/crypto/rustls.rs

```diff
@@ -180,7 +179,8 @@ impl crypto::Session for TlsSession {
     ) -> Result<(), ExportKeyingMaterialError> {
         self.inner
             .export_keying_material(output, label, Some(context))
-            .map_err(|_| ExportKeyingMaterialError)
```

Comment:
It's required in order to have the code build with Rustls v0.21.0-alpha. I think it should be included with the other changes in order to keep each commit building/passing tests.

The upstream `export_keying_material` return type changed from `Result((), ...)` to `Result<T, ...>` in [38fdd952bec1ca9ab7c3b25b60026954115561a9](https://github.com/rustls/rustls/commit/38fdd952bec1ca9ab7c3b25b60026954115561a9) which means this code can't rely on the `Ok(())` being passed through `map_err` anymore.

Reply:
@djc I think doing this might be a little tricky and I want to make sure you think it's worthwhile before charging ahead.

To match the upstream exactly we'll have to add a generic bound `T: <AsMut<[u8]>>` to the `export_keying_material` function defined in the Quinn `Session` trait that the rustls `TlsSession` implements. Doing that breaks the object safety of the `Session` trait, and there are ~8 places where existing Quinn code is boxing `Box<dyn Session>`.

WDYT?

Reply:
Ahhh yeah, let's not do that.